src/Data/Str.php

```diff
@@ -169,4 +169,31 @@ public static function ConvertCase(string $str, $encoding = null)
         // returns original string when mbstring is not active.
         return $str;
     }
+
+    /**
+     * Checks if given string is a valid Base64 encoded string.
+     *
+     * @param string $str String to be tested
+     *
+     * @return bool wether it is a valid base64 encoded string or not
+     */
+    public static function isBase64(string $str): bool
+    {
+        return $str === base64_encode(base64_decode($str));
+    }
+
+    /**
+     * Return only a portion of given string.
+     *
+     * @param string   $str      input string to process
+     * @param int      $start    where to start the cut
+     * @param int|null $length   how many characters to return
+     * @param string   $encoding optional encoding to us
```

Comment:
This `@param` comment annotation should change into `@param string|null   $encoding optional encoding to us`.

Reply:
will change it and fix my typo at the end as well. should be "to use"

Reply:
yes

Reply:
Let's fix so i will merge this PR

Reply:
Done and also added another testcase for the above discussion about optional parameters